### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by jonschlinkert**
@doowb I think this is ready to merge, but I'd really appreciate it if you have a moment to do a code review. thanks!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,144 +7,215 @@
...
+ *   e: {path: 'test/fixtures/e.md'},
+ *   f: {path: 'test/fixtures/f.md'},
+ * });
+ * loader.load([{
+ *   g: {path: 'test/fixtures/g.md'},
+ *   h: {path: 'test/fixtures/h.md'},
+ *   i: {path: 'test/fixtures/i.md'},
+ * }]);
+ * console.log(loader.cache);
+ * ```
```
Comment (doowb) [inline index.js]: This example is using `.load`, but it's for the `.addViews` method.

**Review by doowb — APPROVED** [review]
These changes look awesome. It's a lot easier to follow. There was only one comment about one of the examples.

I really like how you're getting the glob parent for each pattern in the array and handling when a pattern isn't a glob.

File: index.js (hunk 1 above)
Comment (jonschlinkert) [inline index.js]: fixed, thanks!

**Comment by jonschlinkert** [comment]
thanks @doowb! really appreciate it!

**Pull request merged**